[Style] Use strong Style value types for the translate, rotate, scale, and perspective properties

weinig opened the pull request:
#### f7b8b7ad961e1f2c09f06751e2e99b390ffd4798
<pre>
[Style] Use strong Style value types for the translate, rotate, scale, and perspective properties
<a href="https://bugs.webkit.org/show_bug.cgi?id=294526">https://bugs.webkit.org/show_bug.cgi?id=294526</a>

Reviewed by Darin Adler.

Add strongly typed wrappers for the translate, rotate, scale,
and perspective properties. These are all &quot;zero-cost&quot; wrappers
around the existing reference counted platform transform
operation types or around a Length in the case of perspective.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/animation/BlendingKeyframes.cpp:
* Source/WebCore/animation/KeyframeEffectStack.cpp:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/platform/animation/AcceleratedEffectValues.cpp:
* Source/WebCore/platform/graphics/transforms/TransformOperation.h:
* Source/WebCore/platform/graphics/transforms/TranslateTransformOperation.cpp:
* Source/WebCore/platform/graphics/transforms/TranslateTransformOperation.h:
* Source/WebCore/rendering/RenderLayerBacking.cpp:
* Source/WebCore/rendering/RenderLayerCompositor.cpp:
* Source/WebCore/rendering/RenderLayerModelObject.cpp:
* Source/WebCore/rendering/style/RenderStyle.cpp:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/rendering/style/RenderStyleSetters.h:
* Source/WebCore/rendering/style/StyleRareNonInheritedData.cpp:
* Source/WebCore/rendering/style/StyleRareNonInheritedData.h:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleBuilderCustom.h:
* Source/WebCore/style/StyleExtractorConverter.h:
* Source/WebCore/style/StyleExtractorCustom.h:
* Source/WebCore/style/StyleInterpolationFunctions.h:
* Source/WebCore/style/StyleInterpolationWrappers.h:
* Source/WebCore/style/TransformOperationsBuilder.cpp:
* Source/WebCore/style/TransformOperationsBuilder.h:
* Source/WebCore/style/values/primitives/StylePrimitiveNumericAdaptors.h: Added.
* Source/WebCore/style/values/transforms/StylePerspective.cpp: Added.
* Source/WebCore/style/values/transforms/StylePerspective.h: Added.
* Source/WebCore/style/values/transforms/StyleRotate.cpp: Added.
* Source/WebCore/style/values/transforms/StyleRotate.h: Added.
* Source/WebCore/style/values/transforms/StyleScale.cpp: Added.
* Source/WebCore/style/values/transforms/StyleScale.h: Added.
* Source/WebCore/style/values/transforms/StyleTransformOperationWrapper.h: Added.
* Source/WebCore/style/values/transforms/StyleTranslate.cpp: Added.
* Source/WebCore/style/values/transforms/StyleTranslate.h: Added.
* Source/WebCore/svg/SVGGraphicsElement.cpp:

Canonical link: <a href="https://commits.webkit.org/296320@main">https://commits.webkit.org/296320@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2682060857b2de187f9cbe239dae980d9242ef70

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/108186 "18 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/27849 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/18269 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/113397 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/58669 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/110149 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/28549 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/36399 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/82148 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/58669 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/111134 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/22622 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/97453 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/62579 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/22036 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/15589 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/58134 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91988 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/15645 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/116523 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/35252 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/25957 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/91174 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/35626 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/93729 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/90969 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23176 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35856 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13615 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/31034 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/35151 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/40707 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/34885 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/38243 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/36552 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/platform/animation/AcceleratedEffectValues.cpp:
```diff
@@ -94,7 +94,7 @@ AcceleratedEffectValues AcceleratedEffectValues::clone() const
 }
 
-static LengthPoint nonCalculatedLengthPoint(LengthPoint lengthPoint, const IntSize& borderBoxSize)
+static LengthPoint resolveCalculateValuesFor(const LengthPoint& lengthPoint, const IntSize& borderBoxSize)
```
For performance it seems `IntSize` should probably be passed by value, but `LengthPoint` passed by reference. But for some reason it was the opposite?

weinig replied:
No good reason. Fixed the IntSize though.

darinadler commented on Source/WebCore/rendering/style/RenderStyle.cpp:
```diff
@@ -845,9 +846,9 @@ static bool rareDataChangeRequiresLayout(const StyleRareNonInheritedData& first,
-        || !arePointingToEqualData(first.translate, second.translate))
+    if (first.scale != second.scale
+        || first.rotate != second.rotate
+        || first.translate != second.translate)
```
```suggestion
    if (first.scale != second.scale || first.rotate != second.rotate || first.translate != second.translate)
```
It’s only three, so maybe can be horizontal.

weinig committed:
[Style] Use strong Style value types for the translate, rotate, scale, and perspective properties
https://bugs.webkit.org/show_bug.cgi?id=294526

Reviewed by Darin Adler.

Add strongly typed wrappers for the translate, rotate, scale,
and perspective properties. These are all "zero-cost" wrappers
around the existing reference counted platform transform
operation types or around a Length in the case of perspective.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/animation/BlendingKeyframes.cpp:
* Source/WebCore/animation/KeyframeEffectStack.cpp:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/platform/animation/AcceleratedEffectValues.cpp:
* Source/WebCore/platform/graphics/transforms/TransformOperation.h:
* Source/WebCore/platform/graphics/transforms/TranslateTransformOperation.cpp:
* Source/WebCore/platform/graphics/transforms/TranslateTransformOperation.h:
* Source/WebCore/rendering/RenderLayerBacking.cpp:
* Source/WebCore/rendering/RenderLayerCompositor.cpp:
* Source/WebCore/rendering/RenderLayerModelObject.cpp:
* Source/WebCore/rendering/style/RenderStyle.cpp:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/rendering/style/RenderStyleSetters.h:
* Source/WebCore/rendering/style/StyleRareNonInheritedData.cpp:
* Source/WebCore/rendering/style/StyleRareNonInheritedData.h:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleBuilderCustom.h:
* Source/WebCore/style/StyleExtractorConverter.h:
* Source/WebCore/style/StyleExtractorCustom.h:
* Source/WebCore/style/StyleInterpolationFunctions.h:
* Source/WebCore/style/StyleInterpolationWrappers.h:
* Source/WebCore/style/TransformOperationsBuilder.cpp:
* Source/WebCore/style/TransformOperationsBuilder.h:
* Source/WebCore/style/values/primitives/StylePrimitiveNumericAdaptors.h: Added.
* Source/WebCore/style/values/transforms/StylePerspective.cpp: Added.
* Source/WebCore/style/values/transforms/StylePerspective.h: Added.
* Source/WebCore/style/values/transforms/StyleRotate.cpp: Added.
* Source/WebCore/style/values/transforms/StyleRotate.h: Added.
* Source/WebCore/style/values/transforms/StyleScale.cpp: Added.
* Source/WebCore/style/values/transforms/StyleScale.h: Added.
* Source/WebCore/style/values/transforms/StyleTransformOperationWrapper.h: Added.
* Source/WebCore/style/values/transforms/StyleTranslate.cpp: Added.
* Source/WebCore/style/values/transforms/StyleTranslate.h: Added.
* Source/WebCore/svg/SVGGraphicsElement.cpp:

Canonical link: https://commits.webkit.org/296320@main
